crates/common/fork_choice/lean/src/store.rs

```diff
@@ -728,3 +730,307 @@ impl Store {
+    use ream_post_quantum_crypto::leansig::signature::Signature;
+    use ream_storage::{
+        db::{ReamDB, lean::LeanDB},
+        tables::{field::REDBField, table::REDBTable},
+    };
+    use ssz_types::{VariableList, typenum::U4096};
+    use tempdir::TempDir;
+    use tree_hash::TreeHash;
+
+    use super::Store;
+    use crate::genesis::setup_genesis;
+
+    pub fn db_setup() -> LeanDB {
+        let temp_dir = TempDir::new("lean_test").unwrap();
+        let temp_path = temp_dir.path().to_path_buf();
+        let ream_db = ReamDB::new(temp_path).expect("unable to init Ream Database");
+        ream_db.init_lean_db().unwrap()
+    }
+
+    pub async fn sample_store(no_of_validators: usize) -> (Store, LeanState) {
+        let (genesis_block, genesis_state) =
+            setup_genesis(0, generate_default_validators(no_of_validators));
+
+        let genesis_block_hash = genesis_block.tree_hash_root();
+
+        let checkpoint = Checkpoint {
+            slot: genesis_block.slot,
+            root: genesis_block_hash,
+        };
+        let signed_genesis_block = build_signed_block_with_attestation(
+            AttestationData {
+                slot: genesis_block.slot,
+                head: checkpoint,
+                target: checkpoint,
+                source: checkpoint,
+            },
+            &genesis_block,
+            &mut VariableList::default(),
+        );
+
+        set_lean_network_spec(LeanNetworkSpec::ephemery().into());
+
+        let db = db_setup();
+        (
+            Store::get_forkchoice_store(signed_genesis_block, genesis_state.clone(), db, None)
+                .unwrap(),
+            genesis_state,
+        )
+    }
+
+    pub fn build_signed_attestation(
+        validator: u64,
+        slot: u64,
+        head: Option<Checkpoint>,
+        source: Option<Checkpoint>,
+        target: Option<Checkpoint>,
+    ) -> SignedAttestation {
+        let data = Attestation {
+            validator_id: validator,
+            data: AttestationData {
+                slot,
+                head: head.unwrap_or_default(),
+                target: target.unwrap_or_default(),
+                source: source.unwrap_or_default(),
+            },
+        };
+        SignedAttestation {
+            message: data,
+            signature: Signature::blank(),
+        }
+    }
+
+    pub fn build_signed_block_with_attestation(
+        attestation_data: AttestationData,
+        block: &Block,
+        signatures: &mut VariableList<Signature, U4096>,
+    ) -> SignedBlockWithAttestation {
+        let attestation = Attestation {
+            validator_id: block.proposer_index,
+            data: attestation_data,
+        };
+
+        signatures.push(Signature::blank()).unwrap();
+        SignedBlockWithAttestation {
+            message: BlockWithAttestation {
+                block: block.clone(),
+                proposer_attestation: attestation,
+            },
+            signature: signatures.clone(),
```

Comment:
```suggestion
    pub fn build_signed_block_with_attestation(
        attestation_data: AttestationData,
        block: &Block,
        signatures: mut VariableList<Signature, U4096>,
    ) -> SignedBlockWithAttestation {
        let attestation = Attestation {
            validator_id: block.proposer_index,
            data: attestation_data,
        };

        signatures.push(Signature::blank()).unwrap();
        SignedBlockWithAttestation {
            message: BlockWithAttestation {
                block: block.clone(),
                proposer_attestation: attestation,
            },
            signature: signatures,
```

We don't reuse the signatures so we can just make `&mut` to `mut` and avoid the clone

Reply:
You mean
```
mut signatures: VariableList<Signature, U4096>,
```
right?